### PR TITLE
Begin adding Wildfly deployment to INSTALL.md

### PR DESCRIPTION
**Pull request by cecilia-donnelly**
Based on https://groups.google.com/forum/#!topic/psm-dev/nzC3pP1XMFg, by
@jasonaowen.

I'm a little unclear on what we need to do in Step 4.  Maybe @jasonaowen can clarify?

This is related to issue #5.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -83,6 +83,71 @@ this set up, based on the info in the /docs subdir.
 Several people have done work to get this up and running so far.  This
 is a summary of their notes.
 
+### Wildfly Version
+
+See the [WIP
+branch](https://github.com/OpenTechStrategies/psm/tree/jasonaowen/wip-wildfly).
```
Comment (jasonaowen) [inline INSTALL.md]: Now that this is the branch being added to, I don't think we need to refer to it!

File: INSTALL.md [hunk 2]
```diff
@@ -83,6 +83,71 @@ this set up, based on the info in the /docs subdir.
...
+3. Build the application with `ant`.
+
+       $ cd ../psm/psm-app
+       
+       # create an EAR file:
+       $ ant deploy
+       
+       # This depends on libraries provided by the application server, which
+       # are pulled in as part of the [wip-wildfly
+       # branch](https://github.com/OpenTechStrategies/psm/tree/jasonaowen/wip-wildfly).
```
Comment (jasonaowen) [inline INSTALL.md]: Same - delete reference to branch.

File: INSTALL.md [hunk 3]
```diff
@@ -83,6 +83,71 @@ this set up, based on the info in the /docs subdir.
...
+
+4. Add the deploy target (?).  "The current deploy target is inside the
+   WildFly home (also something to change later), but you can add it
+   through the management interface: Deployments > Add > Upload a new
+   deployment > browse to file."
```
Comment (jasonaowen) [inline INSTALL.md]: Hopefully building will make this clear, but currently `ant deploy` puts its output artifact in `wildfly-10.1.0.Final/server/default/deploy/cms-portal-services.ear`. You can upload that artifact into the WildFly management UI.

File: INSTALL.md [hunk 4]
```diff
@@ -83,6 +83,71 @@ this set up, based on the info in the /docs subdir.
...
+            </mail-session>
+            <mail-session name="default" jndi-name="java:jboss/mail/Default">
+                <smtp-server outbound-socket-binding-ref="mail-smtp"/>
+            </mail-session>
+        </subsystem>
+
+        <outbound-socket-binding name="mail-smtp">
+          - <remote-destination host="localhost" port="25"/>
+          + <remote-destination host="localhost" port="1025"/>
+        </outbound-socket-binding>
```
Comment (jasonaowen) [inline INSTALL.md]: Today I learned these can be configured in the UI! Do you think we should document that instead? Or maybe just commit an example dev `standalone-full.xml`?

Comment (cecilia-donnelly) [inline INSTALL.md]: I'd prefer to commit an example file and change this to `cp standalone-full.xml.example path/to/wildfly/standalone/configuration/standalone-full.xml`.  My only concern would be licensing (maybe an abundance of caution!).  @jvasile, can we bring a lightly modified file from Wildfly into this project?  Wildfly is LGPL 2.1.

Comment (jvasile) [inline INSTALL.md]: Based on IRC chat w/ @jasonaowen, I understand this file to be used solely by Wildfly.  None of the PSM stuff interacts with it.  There's no real good reason it can't keep its own license, so bring it in to the PSM, but note in the file that it is LGPL, not Apache.  This also means we have to ship a copy of LGPL, so add a section to COPYING that contains the license and a note that it applies to this one file.  Place upstream copyright notices there too.

Comment (jasonaowen) [inline INSTALL.md]: Thank you, @jvasile!

@cecilia-donnelly I suggest we do that in a separate PR; I think this one is ready to be merged!

Comment (cecilia-donnelly) [inline INSTALL.md]: thanks @jvasile and @jasonaowen!
